### ISSUE TITLE
Suppressing CssSyntaxError for inline styles in svelte files

### ISSUE THREAD

**Issue by mantismamita** [stylelint/stylelint#5685]
### What is the problem you're trying to solve?

Hello, I am currently trying to implement stylelint in a project that uses svelte and several postcss plugins. The problem occurs in the parsing of inline styles in the svelte file. 

We sometimes use the style tag of an html element to define a dynamic custom property as in the following case:

```
<nav
  class="Menu"
  slot="sticky"
  style="
    --nbMenuItems: { displayedItems.length };
    { getStyleUrlVars({ arrowRight }) }"
>
```
The output in this case is 
````
<nav class="Menu svelte-1q3olh0" slot="sticky" style="--nbMenuItems: 3;--arrowRight: url('/client/assets/icon_arrow-2_right-7f4ce63b.svg');">
```
Here I get an error
```
src/components/AnchorMenu/AnchorMenu.svelte
 40:7  ✖  Unknown word   CssSyntaxError
```
I would like to turn off the parsing for inline style tags. Is this possible?

this is also the case with `"defaultSeverity": "warning",` as discussed [here](https://github.com/stylelint/stylelint/issues/2215)

Thanks in advance and for all the hard work that goes into stylelint! 

### What solution would you like to see?

I would like to have an option that allows for parsing only content between <style> tags but otherwise the possibility of doing something like this

```
<nav
   class="Menu"
   slot="sticky"
  style="
    --nbMenuItems: { displayedItems.length };
    /* stylelint-disable-line */
    { getStyleUrlVars({ arrowRight }) }
    /* stylelint-enable-line */
    "
 >
 ```

**Comment by alexander-akait** [comment]
`{ getStyleUrlVars({ arrowRight }) }` is not valid

**Comment by mantismamita** [comment]
Yes, I agree. This is why I want to disable parsing css here. It is in a svelte.js file.

**Comment by jeddy3** [comment]
@mantismamita Thanks for the request and for using the template.

I assume you're using the [postcss-html](https://www.npmjs.com/package/postcss-html) [custom syntax](https://stylelint.io/user-guide/usage/options#customsyntax)? If so, you could reach out to see if they can add an option to their syntax to toggle whether. As this is only a workaround, though, they may hesitate to add it. If they do, we'll need to work out a way of passing options to custom syntaxes here in Stylelint. Perhaps a new `customSyntaxOptions` configuration property.

The ideal solution is [writing custom syntax](https://stylelint.io/developer-guide/syntaxes) specific to Svelte which can parse the construct above. It'll offer you the best linting while avoiding adding options here and in the postcss-html syntax.

**Comment by mantismamita** [comment]
Hi @jeddy3 thank you for the suggestion. I think I managed to find a temporary solution based on your comment. 
```
<!-- postcss-disable -->
    <nav
      class="AnchorMenu"
      slot="sticky"
      style="
      --nbMenuItems: { displayedItems.length };
      { getStyleUrlVars({ arrowRight }) }"
    >
<!-- postcss-enable -->
```

I wouldn't mind writing a custom syntax specific to svelte but it might take a while as I haven't written one before. I'll try following the guide and see how far I get.

**Comment by ota-meshi** [comment]
If you are using a js file for your config, changing it to use `postcss-safe-parser` for CSS parsing with `postcss-html` will make the parser accept some syntax errors. So it may work for you.

```js
{
// ...
overrides: [
  {
    files: ['*.svelte', '**/*.svelte'],
    customSyntax: require('postcss-html')({ css: 'postcss-safe-parser' }),
  }
]
}
```

You may need to explicitly install postcss-safe-parser.

https://github.com/ota-meshi/postcss-html#advanced-use-cases
https://github.com/postcss/postcss-safe-parser

**Comment by kelvindecosta** [comment]
Hey everyone!

The following fix worked for me:

- Install these dependencies:

  ```bash
  npm i -D postcss-html stylelint-config-html
  ```

- Add `stylelint-config-html/svelte` to `extends`

Hope this helps!

**Comment by kelvindecosta** [comment]
Hey everyone!

The following fix worked for me:

- Install these dependencies:

  ```bash
  npm i -D postcss-html stylelint-config-html
  ```

- Add `stylelint-config-html/svelte` to `extends`

Hope this helps!
